articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Login
+description: This tutorial demonstrates how to add user login, logout, and profile to a Node.js Express application.
+budicon: 448
+topics:
+  - quickstarts
+  - webapp
+  - login
+  - user profile
+  - logout
+  - nodejs
+contentType: tutorial
+useCase: quickstart
+github:
+  path: 01-Login
+---
+
+<!-- markdownlint-disable MD002 -->
+<!-- markdownlint-disable MD034 -->
+
+## Getting Started
+In this quickstart you will learn how to implement authentication in an Express application using the [Express OpenID Connect](https://github.com/auth0/express-openid-connect) library, authored and maintained by Auth0.
+
+::: warning
+Express OIDC is still in an [early access stage](/).
+:::
+
+### Install Dependencies
+
+```sh
+# Using NPM
+npm install express-openid-connect express-session
+
+# Using Yarn
+yarn add express-openid-connect express-session
+```
+
+## Configure Auth0
+<%= include('../../../_includes/_callback_url') %>
+<%= include('../../../_includes/_logout_url' }) %>
+
+## Integrate Auth0
+### Express Middleware
+There are three middlewares required to integrate Auth0 in your application.
+
+- `express.urlencoded` - parses urlencoded payloads sent back from the Auth0 server.
+- `express-session` - simple session middleware for Express (this can be swapped with any session middleware for express).
+- `express-openid-connect.auth` - an [OIDC](/protocols/oidc) compliant authentication middleware.
+
+::: note
+To learn more about using middleware with Express view the [official documentation](https://expressjs.com/en/guide/using-middleware.html)
+:::
+
+```js
+const express = require('express');
+const session = require('express-session');
+const { auth } = require('express-openid-connect');
+
+// Example initialization of Express
+const app = express();
+
+// parses urlencoded payloads sent back from the Auth0 server
+app.use(express.urlencoded({
+  extended: false
+}));
+
+// session middleware required to handle user sessions
+app.use(session({
+  secret: 'this should be a secret',
+  resave: true,
+  saveUninitialized: false
+}));
+
+// express-openid-connect.auth
+app.use(auth({
+  required: false,
```

Comment:
I meant to ask earlier ... what do you think about this? The SDK is default "require auth for all routes." Do you think that's the most common case or do you think that applications would only need specific routes protected? If this default was switched, it feels like the DX is nicer as well:

```js
// Auth required everywhere
const { auth } = require('express-openid-connect');

app.use(auth({ 
    required: true 
});

app.get('/profile', (req, res) => {
    // ...
});

// Auth required on specific routes
const { auth, requiresAuth } = require('express-openid-connect');
app.use(auth());
app.get('/profile', requiresAuth(), (req, res) => {
    // ...
});
```

Reply:
It's a fairly minor point but changing this is a pretty big breaking change and one I'd like to get in before EA if we decide to go that direction).

Reply:
Going to punt this to beta. Have not had any complaints or comments about this, just something I was thinking about.

https://auth0team.atlassian.net/browse/SDK-1071